classlib/src/main/java/org/teavm/classlib/java/util/concurrent/atomic/TAtomicLong.java

```diff
@@ -0,0 +1,135 @@
+/*
+ *  Copyright 2018 martin below.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.teavm.classlib.java.util.concurrent.atomic;
+
+import java.io.Serializable;
+import java.util.function.LongBinaryOperator;
+import java.util.function.LongUnaryOperator;
+
+public class TAtomicLong extends Number implements Serializable {
+    private long value;
+
+    public TAtomicLong() {
+    }
+
+    public TAtomicLong(final long value) {
+        this.value = value;
+    }
+
+    public final long get() {
+        return value;
+    }
+
+    public final void set(final long newValue) {
+        value = newValue;
+    }
+
+    public final void lazySet(final long newValue) {
+        value = newValue;
+    }
+
+    public final long getAndSet(final long newValue) {
+        long result = value;
+        value = newValue;
+        return result;
+    }
+
+    public final boolean compareAndSet(final long expect, final long update) {
+        if (value != expect) {
+            return false;
+        }
+        value = update;
+        return true;
+    }
+
+    public final boolean weakCompareAndSet(final long expect, final long update) {
+        return compareAndSet(expect, update);
+    }
+
+    public final long getAndIncrement() {
+        return value++;
+    }
+
+    public final long getAndDecrement() {
+        return value--;
+    }
+
+    public final long getAndAdd(long delta) {
+        long result = value;
+        value += delta;
+        return result;
+    }
+
+    public final long incrementAndGet() {
+        return ++value;
+    }
+
+    public final long decrementAndGet() {
+        return --value;
+    }
+
+    public final long addAndGet(final long delta) {
+        value += delta;
+        return value;
+    }
+
+    public final long getAndUpdate(final LongUnaryOperator updateFunction) {
```

Comment:
FYI. You must have copy-pasted this code from my `TAtomicInteger` implementation. However, both this code and corresponding code in `TAtomicInteger` is wrong. Actually, TeaVM is not single-threaded, context switch can occur as you call any method, and thus this code should be synchronized. I'll accept this code anyway and try to fix it later myself.